### DISCUSSION TITLE
XYPlots now longer resize nicely with with update to 1.5

### DISCUSSION THREAD

**Discussion by Richardsks** [jfree/jfreechart#295] [Q&A] [answered]
Hello,
I am very late to the party as far as updating my freechart version from 1.1.19 to 1.5.3.
No big issues making the API changes and all but one of our apps that use freechart look just as great.

The one issue:
I have a CombinedDomainXYPlot with a common time X Axis. 
That  is inside of a chartPanel that is the viewport to a JScrollPane.  

Prior to upgrading to the newer version of Freechart, the plots within the combined domain plot would resize nicely as I resized the frame until a certain point when the scrollbar would kick in. 

That no longer seems to happen.  The size of the plots and the chart panel stay fixed and I get a scroll bar immediately. 

Any suggestions? Thank you

**Comment by trashgod** [comment]
This seems like more of a Swing problem, possibly layout related. Using v1.5.3, I tried this [example](https://stackoverflow.com/a/11895709/230513) and saw the result you describe as the frame is resized:

    frame.add(new JScrollPane(chartPanel), BorderLayout.CENTER);

[Pane and zoom](https://stackoverflow.com/a/44151032/230513) also seems to work as expected.

**Reply by Richardsks** [reply]
thank you, That's Interesting. I will look at the code in the example. Its weird that the same exact code behaves differently between versions of jfreechart. I just assumed it had to do with jfreechart and not swing.

**Comment by Richardsks** [comment, marked as answer]
Turns out that the change made between JFreechart version causing the behavior to differ was in the JFreechart ChartPanel class. The DEFAULT_MAXIMUM_DRAW_HEIGHT was changed from 768 to 1536 and the DEFAULT_MAXIMUM_DRAW_WIDTH was changed from 1024 to 2048. The increased values made it so that the scaling didn't kick in because it was no longer hitting the max draw values.

**Reply by trashgod** [reply]
Good observation: scrollbars will appear when the scroll pane is smaller than the preferred size of the `ChartPanel`; more on preferred size [here](https://stackoverflow.com/a/10277372/230513).
